src/plugins/dashboard/public/application/actions/clone_panel_action.test.tsx

```diff
@@ -113,56 +135,156 @@ test('Clone adds a new embeddable', async () => {
+  const dashboard = genericEmbeddable.getRoot() as IContainer;
+  const panel = dashboard.getInput().panels[genericEmbeddable.id] as DashboardPanelState;
+  const action = new ClonePanelAction(coreStart);
+  // @ts-ignore
+  const newPanelWithoutId = await action.cloneEmbeddable(panel, genericEmbeddable);
+  expect(coreStart.savedObjects.client.get).toHaveBeenCalledTimes(0);
+  expect(coreStart.savedObjects.client.find).toHaveBeenCalledTimes(0);
+  expect(coreStart.savedObjects.client.create).toHaveBeenCalledTimes(0);
+  expect(newPanelWithoutId.type).toEqual(genericEmbeddable.type);
+});
+
+test('Clones a non-RefOrVal embeddable by reference if the panel has a savedObjectId', async () => {
+  const dashboard = genericEmbeddable.getRoot() as IContainer;
+  const panel = dashboard.getInput().panels[genericEmbeddable.id] as DashboardPanelState;
   panel.explicitInput.savedObjectId = 'holySavedObjectBatman';
   const action = new ClonePanelAction(coreStart);
   // @ts-ignore
-  const newPanel = await action.cloneEmbeddable(panel, embeddable.type);
+  const newPanel = await action.cloneEmbeddable(panel, genericEmbeddable);
   expect(coreStart.savedObjects.client.get).toHaveBeenCalledTimes(1);
   expect(coreStart.savedObjects.client.find).toHaveBeenCalledTimes(1);
   expect(coreStart.savedObjects.client.create).toHaveBeenCalledTimes(1);
-  expect(newPanel.type).toEqual(embeddable.type);
+  expect(newPanel.type).toEqual(genericEmbeddable.type);
 });
 
-test('Gets a unique title ', async () => {
+test('Gets a unique from the saved objects library', async () => {
   coreStart.savedObjects.client.find = jest.fn().mockImplementation(({ search }) => {
-    if (search === '"testFirstTitle"') return { total: 1 };
-    else if (search === '"testSecondTitle"') return { total: 41 };
-    else if (search === '"testThirdTitle"') return { total: 90 };
+    if (search === '"testFirstClone"') {
+      return {
+        savedObjects: [
+          {
+            attributes: { title: 'testFirstClone' },
+            get: jest.fn().mockReturnValue('testFirstClone'),
+          },
+        ],
+        total: 1,
+      };
+    } else if (search === '"testBeforePageLimit"') {
+      return {
+        savedObjects: [
+          {
+            attributes: { title: 'testBeforePageLimit (copy 9)' },
+            get: jest.fn().mockReturnValue('testBeforePageLimit (copy 9)'),
+          },
+        ],
+        total: 10,
+      };
+    } else if (search === '"testMaxLogic"') {
+      return {
+        savedObjects: [
+          {
+            attributes: { title: 'testMaxLogic (copy 10000)' },
+            get: jest.fn().mockReturnValue('testMaxLogic (copy 10000)'),
+          },
+        ],
+        total: 2,
+      };
+    } else if (search === '"testAfterPageLimit"') {
+      return { total: 11 };
+    }
   });
+
+  const action = new ClonePanelAction(coreStart);
+  // @ts-ignore
```

Comment:
Nit: These ts-ignores are for ignoring calling private methods, correct? Is there a way to disable the specific rule for all of these lines, rather than individually ts-ignoring each line?

Reply:
Looked in to this while making these tests and nope, that doesn't exist (although it has been requested many, many times on the TypeScript forums 😛). There *is* the option to `ts-ignore` the **entire file** but not chunks of code.

Reply:
So so strange that they don't have that capability!